core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/NativeStore.java

```diff
@@ -311,14 +335,33 @@ protected void shutDownInternal() throws SailException {
 
 		try {
 			store.close();
-
-			logger.debug("NativeStore shut down");
 		} finally {
 			dirLock.release();
 			if (dependentServiceResolver != null) {
 				dependentServiceResolver.shutDown();
 			}
-			logger.debug("NativeStore shut down");
+		}
+
+		if (isTmpDatadir) {
+			File tmpDatadir = getDataDir();
```

Comment:
Above, in setDatadir, you're checking if the supplied datadir is null to determine if `isTmpDatadir` should be true. I think that means that it's possible to get at this poin with the result `getDataDir()` being null? If so you need a null-check here somewhere.

Reply:
I don't think this is the case (though probably by coincidence: this could be the case when `setDatadir(null)` is called and `shutdownInternal()` is called without initialize... but currently this is not the case because `shutdown()` exits early - without calling `shutdownInternal()` - when not initialized)
 
but I'll add a check just to be sure